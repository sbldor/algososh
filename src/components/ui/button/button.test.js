import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Button component snapshot tests', () => {

  test('Button render with text', () => {
    const btn = renderer.create(<Button text='Text' />).toJSON()
    expect(btn).toMatchSnapshot()
  })

  test('Button render without text', () => {
    const btn = renderer.create(<Button />).toJSON()
    expect(btn).toMatchSnapshot()
  })

  test('Button render disabled', () => {
    const btn = renderer.create(<Button disabled />).toJSON()
    expect(btn).toMatchSnapshot()
  })

  test('Button render with loader', () => {
    const btn = renderer.create(<Button isLoader />).toJSON()
    expect(btn).toMatchSnapshot()
  })

})

describe('Button component logic', () => {

  test('Button with a callback on click', () => {
    window.alert = jest.fn()
    render(<Button text='text' onClick={alert('test-text')} />)
    const btn = screen.getByText('text')
    fireEvent.click(btn)
    expect(window.alert).toHaveBeenCalledWith('test-text')
  })

})