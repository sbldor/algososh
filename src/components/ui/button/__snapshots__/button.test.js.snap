// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot tests Button render disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component snapshot tests Button render with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component snapshot tests Button render with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Text
  </p>
</button>
`;

exports[`Button component snapshot tests Button render without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
